src/vcpkg/base/files.cpp

```diff
@@ -3011,6 +3011,27 @@ namespace vcpkg
                 return false;
             }
 
+            if (options == CopyOptions::update_existing)
+            {
+                WIN32_FILE_ATTRIBUTE_DATA attributes_source;
```

Comment:
What about ELF(Linux) and Mach-O(OSX)?

Reply:
Similar code:
ELF: https://github.com/Kitware/CMake/blob/f791cd23d408ff7af104d119898c4dd8e1e87688/Source/cmELF.cxx
Mach-O: https://github.com/Kitware/CMake/blob/f791cd23d408ff7af104d119898c4dd8e1e87688/Source/cmMachO.cxx

Reply:
> Similar code:

And that's why I suggested a cross-platform cmake script.

Reply:
> And that's why I suggested a cross-platform cmake script.

This needs to be usable from MSBuild where we don't have CMake available. That's part of the reason we are in this place in the first place: the current mechanism needs to defend against a zillion different powershell versions because the one vcpkg wants to supply is not known to be present.

Reply:
> What about ELF(Linux) and Mach-O(OSX)?

`applocal.ps1` doesn't handle those so they're not in scope here.

Reply:
Also note that CMake's implementation depends on needing to find dumpbin and is vulnerable to breaking on localized systems.

Reply:
> This needs to be usable from MSBuild where we don't have CMake available.

Just for understanding: When msbuild is able to call vcpkg to get the runtime dependencies, why is vcpkg unable to call cmake itself to get the runtime dependencies. I mean cmake must be available for vcpkg otherwise it could not build a single port.

Reply:
>I mean cmake must be available for vcpkg otherwise it could not build a single port.

In classic mode (and "export" et al.) vcpkg does not build ports when called by msbuild.